components/spiffs/esp_spiffs.c

```diff
@@ -764,14 +881,89 @@ static void vfs_spiffs_seekdir(void* ctx, DIR* pdir, long offset)
 
 static int vfs_spiffs_mkdir(void* ctx, const char* name, mode_t mode)
 {
+#ifdef CONFIG_SPIFFS_USE_DIR
+    assert(name);
+    esp_spiffs_t * efs = (esp_spiffs_t *)ctx;
+
+    int fd = SPIFFS_open(efs->fs, name, SPIFFS_CREAT | SPIFFS_WRONLY, 0);
+    if (fd < 0) {
+        errno = spiffs_res_to_errno(SPIFFS_errno(efs->fs));
+        SPIFFS_clearerr(efs->fs);
+        return -1;
+    }
+    vfs_spiffs_update_meta(efs->fs, fd, SPIFFS_TYPE_DIR);
+
+    if (SPIFFS_close(efs->fs, fd) < 0) {
+        errno = spiffs_res_to_errno(SPIFFS_errno(efs->fs));
+        SPIFFS_clearerr(efs->fs);
+        return -1;
+    }
+    return 0;
+#else
     errno = ENOTSUP;
     return -1;
+#endif
```

Comment:
I left the comment near the end of `vfs_spiffs_rmdir` function, please do the same here to put the shorter `#if` branch first, and longer one after it.